configuration/postgres/postgres.go

```diff
@@ -0,0 +1,296 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/dapr/components-contrib/configuration"
+	"github.com/dapr/kit/logger"
+	"github.com/google/uuid"
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4/pgxpool"
```

Comment:
pgx v5 is about to come out (currently it's in a RC-like state)... any chance you could update to that one right away? we'll have to upgrade eventually anyways.

Reply:
Agree. Since Postgres configuration will be in Alpha will move to pgx v5

Reply:
Checked on v5 and it is still in beta.So proceeding with v4

Reply:
I agree with @mukundansundar . This is a preview component so it's fine with using v5 right away. We'll have to upgrade eventually anyways...

Reply:
The only caveat that I have is do we know eta... Will it be longer than the eta we plan for making this component stable?

Reply:
> The only caveat that I have is do we know eta... Will it be longer than the eta we plan for making this component stable?

pgx AFAIK is maintained by volunteers, but it's been in beta for a while already and I doubt it will be much longer.

Reply:
Actually released 2days ago.. https://github.com/jackc/pgx/releases/tag/v5.0.0

Reply:
updated to v5 with few minor updates to connect and waitForNotification